portality/models/oaipmh.py

```diff
@@ -71,9 +71,8 @@ def list_records(self, from_date=None, until_date=None, oai_set=None, list_size=
         if start_after is not None or from_date is not None or until_date is not None or oai_set is not None:
 
             if oai_set is not None:
-                s = deepcopy(self.set_limit)
-                s["term"]["index.schema_subject.exact"] = oai_set
-                q["query"]["bool"]["must"].append(s)
+                a = oai_set.replace(constants.SUBJECTS_SCHEMA,"")
+                q["query"]["bool"]["should"] = {"match":{"index.classification": a}}
```

Comment:
I think this should be on `index.classification.exact` ? Or does changing it from a `term` query to a `match` fix that?

Also, are we sure it should be a `should` query?  Seems like is should be a `must` query like the one it is replacing?

Reply:
Yes, I'm confident that this should be a `term` query and a `must`.

Can we please lay it out the way way as it was before, so use `self.set_limit` and just modify it to use `index.classification.exact` instead.

Reply:
Yes, you're right. The `should` and `match` is the only way to make it work without the `exact` filter. When `exact` is applied, then it is `must` and `term` that works.